unlock-app/src/services/web3Service.js

```diff
@@ -496,7 +498,8 @@ export default class Web3Service extends EventEmitter {
           args[input.name] = decoded[input.name]
           return args
         }, {})
-        this.emitContractEvent(transactionHash, event.name, args)
+
+        this.emitContractEvent(transactionHash, log.address, event.name, args)
```

Comment:
Since we are not passing the transaction around, we need to keep track of what smart contract was actually triggering that event.